Tests/ConfigurationManagerSpec.swift

```diff
@@ -97,6 +97,24 @@ class ConfigurationManagerSpec: QuickSpec {
                     configurationManager.fetchAndSaveConfigData(completion: { _ in })
                     expect(configurationManager.scheduledTask).toEventuallyNot(beNil())
                 }
+
+                it("should not retry for .missingOrInvalidSubscriptionId error") {
+                    configurationService.simulateRequestFailure = true
+                    configurationService.mockedError = .missingOrInvalidSubscriptionId
+                    configurationManager.fetchAndSaveConfigData(completion: { _ in })
+                    expect(configurationManager.scheduledTask).toAfterTimeout(beNil())
+                }
+
+                it("should return disable response for .missingOrInvalidSubscriptionId error") {
+                    configurationService.simulateRequestFailure = true
```

Comment:
Maybe the simulateRequestFailure = true can be factored out all of these tests into a beforeEach inside a describe("request failure") block?